### PR TITLE
mess with tracing

### PR DESCRIPTION
**Pull request by AlexTugarev**
Trying to get rid of the millions of error logged in prod. 

- [x] /werft with-observability=true
- [x] /werft with-helm=true
- [x] /werft with-clean-slate-deployment

```release-notes
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@AlexTugarev: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/server/src/workspace/messagebus-integration.ts [hunk 1]
```diff
@@ -98,11 +98,14 @@ export class PrebuildUpdatableQueueListener implements MessagebusListener {
         }
 
         const spanCtx = opentracing.globalTracer().extract(opentracing.FORMAT_HTTP_HEADERS, message.properties.headers);
-        const span = !!spanCtx
-            ? opentracing.globalTracer().startSpan(`/messagebus/${MessageBusHelperImpl.PREBUILD_UPDATABLE_QUEUE}`, {
-                  references: [opentracing.childOf(spanCtx!)],
-              })
-            : undefined;
+        let span;
+        if (!!spanCtx && !!spanCtx.toTraceId()) {
```
Comment (AlexTugarev) [inline components/server/src/workspace/messagebus-integration.ts:102]: @geropl, I believe that's the root of the issue. `traceId` was missing.

File: components/gitpod-messagebus/src/messagebus.ts [hunk 2]
```diff
@@ -508,17 +530,22 @@ export abstract class AbstractTopicListener<T> implements MessagebusListener {
...
         } catch (e) {
-            log.warn('Caught message without or with invalid JSON content', e, { message });
+            log.warn("Caught message without or with invalid JSON content", e, { message });
         }
 
         if (msg) {
             const spanCtx = globalTracer().extract(FORMAT_HTTP_HEADERS, message.properties.headers);
-            const span = !!spanCtx ? globalTracer().startSpan(`/messagebus/${this.exchangeName}`, { references: [childOf(spanCtx!)] }) : undefined;
+            let span;
+            if (!!spanCtx && !!spanCtx.toTraceId()) {
```
Comment (AlexTugarev) [inline components/gitpod-messagebus/src/messagebus.ts:539]: `traceId` was missing here as well..

File: components/gitpod-protocol/src/util/tracing.ts [hunk 3]
```diff
@@ -22,9 +22,19 @@ export type TraceContextWithSpan = TraceContext & {
 export namespace TraceContext {
     export function startSpan(operation: string, parentCtx?: TraceContext): opentracing.Span {
         const options: opentracing.SpanOptions = {};
-        if (parentCtx && parentCtx.span && !!parentCtx.span.context().toSpanId()) {
-            options.childOf = parentCtx.span;
+
+        // try out: https://github.com/jaegertracing/jaeger-client-node/issues/432
+        if (!!parentCtx?.span) {
+            const ctx = parentCtx?.span?.context();
+            if (ctx && !!ctx.toTraceId()) {
```
Comment (AlexTugarev) [inline components/gitpod-protocol/src/util/tracing.ts:29]: `traceId` was missing here as well.

Comment (AlexTugarev) [reply components/gitpod-protocol/src/util/tracing.ts:29]: wait, maybe not here 🤔

**Comment by AlexTugarev** [comment]
@geropl or @mads-hartmann, please help to get this verified.

File: components/gitpod-protocol/src/util/tracing.ts (hunk 3 above)
Comment (geropl) [reply components/gitpod-protocol/src/util/tracing.ts:29]: Also we had a check for `!!parentCtx.span.context().toSpanId()` before, I think we should keep it in as well.

**Comment by geropl** [comment]
@AlexTugarev Thanks a ton for fixing this - the traces indeed look complete!

Please have a look at [this comment](https://github.com/gitpod-io/gitpod/pull/9281#discussion_r850229739). And then I'm happy to :heavy_check_mark: :checkered_flag: .

File: components/gitpod-protocol/src/util/tracing.ts (hunk 3 above)
Comment (AlexTugarev) [reply components/gitpod-protocol/src/util/tracing.ts]: done

**Review by easyCZ — COMMENTED** [review]
The lint here obscures the nuance of the change. It'd make it easier for reviewers to first lint the file (can be done with `pre-commit run --file <file>`) and then base this on top, to highlight the changes made to tracing.

**Comment by AlexTugarev** [comment]
@easyCZ, you are absolutely right. 😿

**Review by geropl — APPROVED** [review]

**Pull request merged**